### PR TITLE
.NET Interactive: Add_try_dotnet_batch_31a

### PR DESCRIPTION
**Pull request by v-mepa**
fyi @mairaw, I am going back to creating multiple PR's for each batch. You can review them as per your convenience. 

## Add_try_dotnet_batch_31a

S No | URL | PageViews | BounceRate | Try .NET Status | Comments
-- | -- | -- | -- | -- | --
1 | https://docs.microsoft.com/en-us/dotnet/api/system.convert.todecimal | 8945 | 14 | Batch 31 | Except 3, 7, 8, 12, 15, 17,   18. Not runnable or depend on each other
2 | https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.numberformat | 707 | 9 | Batch 31 |  
3 | https://docs.microsoft.com/en-us/dotnet/api/system.random.sample | 697 | 9 | Batch 31 |  
4 | https://docs.microsoft.com/en-us/dotnet/api/system.type.isclass | 676 | 14 | Batch 31 |  
5 | https://docs.microsoft.com/en-us/dotnet/api/system.bitconverter.toint16 | 667 | 13 | Batch 31 |

### CODE REVIEW

File: xml/System/Convert.xml [hunk 1]
```diff
@@ -6726,9 +6726,9 @@
 ## Examples
  The following example illustrates the use of `ToDecimal`. It attempts to convert a <xref:System.String> to a <xref:System.Decimal>, and throws the possible exceptions that may arise during the conversion.
 
- [!code-cpp[System.Convert Snippets#15](~/samples/snippets/cpp/VS_Snippets_CLR_System/system.Convert Snippets/CPP/system.convert snippet.cpp#15)]
- [!code-csharp[System.Convert Snippets#15](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Convert Snippets/CS/system.convert snippet.cs#15)]
- [!code-vb[System.Convert Snippets#15](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Convert Snippets/VB/system.convert snippet.vb#15)]
+ :::code language="cpp" source="~/samples/snippets/cpp/VS_Snippets_CLR_System/system.Convert Snippets/CPP/system.convert snippet.cpp" id="Snippet15":::
+ :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Convert Snippets/CS/system.convert snippet.cs" interactive="try-dotnet-method" id="Snippet15":::
```
Comment (mairaw) [inline xml/System/Convert.xml:6730]:
```suggestion
 :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Convert Snippets/CS/system.convert snippet.cs" id="Snippet15":::
```

File: xml/System/Convert.xml [hunk 2]
```diff
@@ -6986,13 +6986,13 @@
 ## Examples
  The following example defines a `Temperature` class that implements the <xref:System.IConvertible> interface.
 
- [!code-csharp[System.Convert.ToDecimal2#10](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.convert.todecimal2/cs/todecimal2.cs#10)]
- [!code-vb[System.Convert.ToDecimal2#10](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.convert.todecimal2/vb/todecimal2.vb#10)]
+ :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.convert.todecimal2/cs/todecimal2.cs" interactive="try-dotnet" id="Snippet10":::
```
Comment (mairaw) [inline xml/System/Convert.xml:6989]:
```suggestion
 :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.convert.todecimal2/cs/todecimal2.cs" id="Snippet10":::
```

File: xml/System/Convert.xml [hunk 3]
```diff
@@ -6986,13 +6986,13 @@
...
+ :::code language="vb" source="~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.convert.todecimal2/vb/todecimal2.vb" id="Snippet10":::
 
  The following example shows that when a `Temperature` object is passed as a parameter to the <xref:System.Convert.ToDecimal%28System.Object%2CSystem.IFormatProvider%29> method, the <xref:System.IConvertible.ToDecimal%2A?displayProperty=nameWithType> implementation of the `Temperature` class is called to perform the conversion.
 
- [!code-csharp[System.Convert.ToDecimal2#11](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.convert.todecimal2/cs/todecimal2.cs#11)]
- [!code-vb[System.Convert.ToDecimal2#11](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.convert.todecimal2/vb/todecimal2.vb#11)]
+ :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.convert.todecimal2/cs/todecimal2.cs" interactive="try-dotnet-method" id="Snippet11":::
```
Comment (mairaw) [inline xml/System/Convert.xml:6994]:
```suggestion
 :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.convert.todecimal2/cs/todecimal2.cs" id="Snippet11":::
```

File: xml/System/Convert.xml [hunk 4]
```diff
@@ -6457,9 +6457,9 @@
 ## Examples
  The following example converts an <xref:System.Int64> value to a <xref:System.Decimal> value.
 
- [!code-cpp[System.Convert Snippets#10](~/samples/snippets/cpp/VS_Snippets_CLR_System/system.Convert Snippets/CPP/system.convert snippet.cpp#10)]
- [!code-csharp[System.Convert Snippets#10](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Convert Snippets/CS/system.convert snippet.cs#10)]
- [!code-vb[System.Convert Snippets#10](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Convert Snippets/VB/system.convert snippet.vb#10)]
+ :::code language="cpp" source="~/samples/snippets/cpp/VS_Snippets_CLR_System/system.Convert Snippets/CPP/system.convert snippet.cpp" id="Snippet10":::
+ :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Convert Snippets/CS/system.convert snippet.cs" interactive="try-dotnet-method" id="Snippet10":::
```
Comment (mairaw) [inline xml/System/Convert.xml:6461]:
```suggestion
 :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Convert Snippets/CS/system.convert snippet.cs" id="Snippet10":::
```

File: xml/System/Convert.xml [hunk 5]
```diff
@@ -6274,17 +6274,17 @@
...
 
 
 ## Examples
  The following example converts a <xref:System.Double> value to a <xref:System.Decimal> value.
 
- [!code-cpp[System.Convert Snippets#5](~/samples/snippets/cpp/VS_Snippets_CLR_System/system.Convert Snippets/CPP/system.convert snippet.cpp#5)]
- [!code-csharp[System.Convert Snippets#5](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Convert Snippets/CS/system.convert snippet.cs#5)]
- [!code-vb[System.Convert Snippets#5](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Convert Snippets/VB/system.convert snippet.vb#5)]
+ :::code language="cpp" source="~/samples/snippets/cpp/VS_Snippets_CLR_System/system.Convert Snippets/CPP/system.convert snippet.cpp" id="Snippet5":::
+ :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Convert Snippets/CS/system.convert snippet.cs" interactive="try-dotnet-method" id="Snippet5":::
```
Comment (mairaw) [inline xml/System/Convert.xml:6286]:
```suggestion
 :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Convert Snippets/CS/system.convert snippet.cs" id="Snippet5":::
```

File: xml/System/Convert.xml [hunk 6]
```diff
@@ -6124,9 +6124,9 @@
 ## Examples
  The following example attempts to convert a <xref:System.Char> value to <xref:System.Decimal>, and throws <xref:System.InvalidCastException> upon failure.
 
- [!code-cpp[System.Convert Snippets#17](~/samples/snippets/cpp/VS_Snippets_CLR_System/system.Convert Snippets/CPP/system.convert snippet.cpp#17)]
- [!code-csharp[System.Convert Snippets#17](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Convert Snippets/CS/system.convert snippet.cs#17)]
- [!code-vb[System.Convert Snippets#17](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Convert Snippets/VB/system.convert snippet.vb#17)]
+ :::code language="cpp" source="~/samples/snippets/cpp/VS_Snippets_CLR_System/system.Convert Snippets/CPP/system.convert snippet.cpp" id="Snippet17":::
+ :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Convert Snippets/CS/system.convert snippet.cs" interactive="try-dotnet-method" id="Snippet17":::
```
Comment (mairaw) [inline xml/System/Convert.xml:6128]:
```suggestion
 :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Convert Snippets/CS/system.convert snippet.cs" id="Snippet17":::
```

**Review by mairaw — CHANGES_REQUESTED** [review]
Thanks @v-mepa. Check https://review.docs.microsoft.com/en-us/dotnet/api/system.convert.todecimal?branch=pr-en-us-5438&view=net-5.0#code-try-32 to see which ones were missed. I commented on the ones that shouldn't have .NET Interactive but there a couple of ones that should be that aren't (review the examples number and let me know if I got the numbers wrong).

**Comment by v-mepa** [comment]
There are 18 examples and I counted the examples not the samples, my mistake. I have updated the missing samples and removed interactive from the ones mentioned in the comment.

**Review by mairaw — APPROVED** [review]

**Pull request merged**